src/components/TicTacToe.tsx

```diff
@@ -25,38 +32,57 @@ class TicTacToe extends React.Component<object, GameState> {
 
   public submitBoardSize = (event: React.FormEvent<HTMLFormElement>): void => {
     event.preventDefault();
-    this.game = new Game(this.state.boardSize);
-    this.setState({
-      board: this.game.board,
-    });
+    if (this.validateBoardSizeInput(this.state.boardSize)) {
+      this.game = new Game(this.state.boardSize);
+      this.setState({
+        board: this.game.board,
+      });
+    } else {
+      this.setState({
+        error: true,
+      });
+    }
   };
 
   public handlePlayerInput = (coords: Coords): void => {
     this.game.place(coords.row, coords.col);
     this.setState({
       board: this.game.board,
     });
+    if (this.game.winner) {
+      this.setState({
+        winner: this.game.winner,
```

Comment:
By adding this you're duplicating this "winner" concept between the game and this component's state. I would either a) just reference `this.game.winner` or b) add some kind of reader method here that is an alias for `this.game.winner`

Reply:
@rylnd I agree this seems like a smell, but I'm not sure how to change it so that state will change, causing the `GameBoard` component to re-render to show that there's been a winner and preventing further user input. Can you explain a bit more?

Reply:
Fixed after our call.